src/Debugger/Breakpoint.php

```diff
@@ -548,30 +581,91 @@ public function validate()
             );
             return false;
         }
+        return true;
+    }
+
+    private function validateSourceLocation()
+    {
+        if (!$this->location) {
+            $this->setError(
+                StatusMessage::REFERENCE_BREAKPOINT_SOURCE_LOCATION,
+                'Invalid breakpoint location'
+            );
+            return false;
+        }
+
+        $path = $this->location->path();
+        $info = new \SplFileInfo($path);
+
+        // Ensure the file exists and is readable
+        if (!$info->isReadable()) {
+            $this->setError(
+                StatusMessage::REFERENCE_BREAKPOINT_SOURCE_LOCATION,
+                'Invalid breakpoint location - File not found or unreadable: $0.',
+                [$path]
+            );
+            return false;
+        }
 
-        if ($this->condition()) {
-            // validate that the condition is ok for debugging
-            try {
-                if (!stackdriver_debugger_valid_statement($this->condition())) {
-                    $this->setError(
-                        StatusMessage::REFERENCE_BREAKPOINT_CONDITION,
-                        'Invalid breakpoint condition - Invalid operations: $0.',
-                        [$this->condition]
-                    );
-                    return false;
-                }
-            } catch (\ParseError $e) {
+        // Ensure the file is a php file
+        if ($info->getExtension() !== "php") {
+            $this->setError(
+                StatusMessage::REFERENCE_BREAKPOINT_SOURCE_LOCATION,
+                'Invalid breakpoint location - Invalid file type: $0.',
+                [$info->getExtension()]
+            );
+            return false;
+        }
+
+        $file = $info->openFile("r");
+        $file->seek($this->location->line() - 1);
+        $line = $file->current();
+
+        // Ensure the line exists and is not a comment in the file
+        if ($line === false ||
+                preg_match('/^\s*[\/\*]+/', $line) ||
```

Comment:
it's hard, but this regex seems to match with the line 4 on the following code:

```php
<?php

echo 8
  * 6;
```

Reply:
If the line starts with `*`, we'll recursively check the previous lines to look for the beginning multiline comment.

Note that it won't find comments like this:
```
/*
  Some comment like this
 */
```

or something like:
```
$foo; /**
       * Some comment
       */
```